java/spring-boot/app/spring-native/src/main/java/io/opentelemetry/example/graal/OpenTelemetryConfig.java

```diff
@@ -0,0 +1,78 @@
+package io.opentelemetry.example.graal;
+
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.contrib.sampler.RuleBasedRoutingSampler;
+import io.opentelemetry.exporter.otlp.http.trace.OtlpHttpSpanExporter;
+import io.opentelemetry.sdk.autoconfigure.spi.AutoConfigurationCustomizerProvider;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+import io.opentelemetry.semconv.ResourceAttributes;
+import io.opentelemetry.semconv.UrlAttributes;
+import java.util.Collections;
+import java.util.Map;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+
+@Configuration
+public class OpenTelemetryConfig {
+
+  @Bean
+  public AutoConfigurationCustomizerProvider otelCustomizer() {
+    return p ->
+        p.addSamplerCustomizer(this::configureSampler)
+            .addSpanExporterCustomizer(this::configureSpanExporter)
+            .addResourceCustomizer(this::configureResource);
+  }
+
+  /** suppress spans for actuator endpoints */
+  private RuleBasedRoutingSampler configureSampler(Sampler fallback, ConfigProperties config) {
+    return RuleBasedRoutingSampler.builder(SpanKind.SERVER, fallback)
+        .drop(UrlAttributes.URL_PATH, "^/actuator")
+        .build();
+  }
+
+  /**
+   * Configuration for the OTLP exporter. This configuration will replace the default OTLP exporter,
+   * and will add a custom header to the requests.
+   */
+  private SpanExporter configureSpanExporter(SpanExporter exporter, ConfigProperties config) {
+    if (exporter instanceof OtlpHttpSpanExporter) {
+      return ((OtlpHttpSpanExporter) exporter).toBuilder().setHeaders(this::headers).build();
```

Comment:
I wonder if we can also use this builder to set the endpoint here as a string rather than through an env var.

Reply:
Curious, is there a reason why that would be preferred?

Reply:
So the install instructions are all the same for the languages and doesn't require people to switch between code and env vars more than needed. It's a static value anyway.